### ISSUE TITLE
Add a configuration setting for auto wrap

### ISSUE THREAD

**Issue by AdamVig** [stkb/Rewrap#87]
I am adding this extension to my dotfiles so that I can use it in on all of my computers. I want to be able to add a setting like so:
```json
"rewrap.autoWrap": true,
```
but no such setting exists. I have read [the docs on the feature](https://github.com/stkb/Rewrap/wiki/Auto-wrap) and [this issue](https://github.com/stkb/Rewrap/issues/81), but neither address my use case.

I want to be able to turn on auto wrap for _all_ present and future instances of VS Code once and forget about it, not have to manually turn it on each time I set up a new copy of VS Code.

I am willing to try to dig in and make a PR for this, but want to get feedback before proceeding.

**Comment by AdamVig** [comment]
I think the feature is useful enough to have enabled by default, but I am not sure that all users would agree.

Why is it not as simple as it first appears to implement the setting?

**Comment by stkb** [comment]
The problem is deciding what the existing 'Toggle Auto-Wrap' command should then do. (When I added the feature I wasn't familiar with it or how useful it was, so I just added it as a toggle command, in the same style as emacs.) 

These are the options I've thought of:

**Option 1**: The toggle command alters the state of the `rewrap.autoWrap.enabled` setting in the user's config. However there are 3 levels of config files that any setting is taken from: global (user), workspace and workspace folder. So you can't just toggle the global setting, because if it's also set in the workspace settings, toggling will have no effect.
So you can say: the command toggles the setting at the lowest level where it's set; i.e: if it's been set in workspace settings, toggle it there, otherwise toggle it in the user settings. It's doable. Though if someone then asks to be able to set it per-language as well, it gets trickier.

**Option 2**: The command maintains its own state, per session. The setting from the config files is then only used to initialize it when starting vscode up. Possibly a cause of confusion when changing the setting has no immediate effect?
**Option 2a**: The same as (2) but the toggled on/off state is just for the *current document* instead. I think this is the most sensible way, though it's a change from how it currently works.

**Option 3**: Remove the command entirely and just go with setting. But I do think there are people who need it, for when working on certain types of text.

 ----

Then you look at all that and think: hmm, or we can just set it **on** by default instead (and and see if someone complains before choosing one of the above; anyone who doesn't like it can just turn it off once anyway).

**Comment by AdamVig** [comment]
I think that option three is the best one.

Most extensions I have encountered are only configurable via settings, so it seems like a better choice because it would make this extension consistent with others in the community.

Short of that, I think enabling it by default would also make sense and would preserve the existing behavior.

**Comment by stkb** [comment]
I think there are definitely cases where people need to be able to toggle it on or off quickly. Lots of people seem to use this extension to help write LaTeX or markdown documents etc, and while the support there is pretty good, there are cases where you just need to turn it off for a moment to write a long line or something (I've had this myself).

I'll publish a version with the new default tomorrow.

**Comment by AdamVig** [comment]
That makes sense. Thanks, that will be helpful for me and for future consumers who want it to "just work" everywhere.

**Comment by stkb** [comment]
OK I changed my mind... I chickened out of turning it on by default :) I felt that for people with no knowledge of the feature, the fact that vscode would be randomly adding newlines as they type--with no explanation--would be too much. This is exacerbated by the fact that it would also happen in code lines for source code files that Rewrap has no knowledge about.

Anyway I worked out a way to do it, which I'll spare you the details of (it's mostly like option 1 above). 

But you will now need to add `"rewrap.autoWrap.enabled": true` to your settings files.

It's in v1.9.1 (just published)
